packages/components/src/Tile/src/TileGroup.tsx

```diff
@@ -0,0 +1,122 @@
+import { useResponsiveValue, useStyledSystem, type ResponsiveProp, type StyledComponentProps } from "@hopper-ui/styled-system";
+import clsx from "clsx";
+import { Children, forwardRef, type CSSProperties, type ForwardedRef } from "react";
+import { Provider, ToggleButtonGroup, useContextProps, type Key } from "react-aria-components";
+
+import { cssModule, type BaseComponentDOMProps } from "../../utils/index.ts";
+
+import { InternalTileContext } from "./TileContext.ts";
+import { TileGroupContext } from "./TileGroupContext.ts";
+
+import styles from "./TileGroup.module.css";
+
+export const GlobalTileGroupCssSelector = "hop-TileGroup";
+
+export interface TileGroupProps extends
+    StyledComponentProps<BaseComponentDOMProps> {
+    /**
+    *  Whether the segmented control is disabled.
+    */
+    isDisabled?: boolean;
+    /**
+     * The id of the currently selected item (controlled).
+     */
+    selectedKey?: Key;
+    /**
+     * The id of the initial selected item (uncontrolled).
+     */
+    defaultSelectedKey?: Key;
+    /**
+     * Handler that is called when the selection changes.
+     */
+    onSelectionChange?: (id: Key) => void;
+    /**
+     * The number of columns to display the tiles in.
+     * @default 3
+     */
+    numberOfColumns?: ResponsiveProp<number>;
+}
+
+const TileGroup = (props: TileGroupProps, ref: ForwardedRef<HTMLDivElement>) => {
+    [props, ref] = useContextProps(props, ref, TileGroupContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children,
+        style,
+        slot,
+        isDisabled,
+        selectedKey,
+        defaultSelectedKey,
+        onSelectionChange,
+        numberOfColumns = 3,
+        ...otherProps
+    } = ownProps;
+
+    const columns = useResponsiveValue(numberOfColumns);
+
+    const classNames = clsx(
+        GlobalTileGroupCssSelector,
+        cssModule(
+            styles,
+            "hop-TileGroup"
+        ),
+        stylingProps.className,
+        className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...style,
+        ...{
+            gridTemplateColumns: `repeat(${columns}, 1fr)`
+        },
+        ...stylingProps.style
+    };
+
+    const onChange = (values: Set<Key>) => {
+        const [firstKey] = values;
+
+        if (!firstKey) {
+            return;
+        }
+
+        onSelectionChange?.(firstKey);
+    };
+
+    if (Children.count(children) < 2) {
```

Comment:
unecessary check. 

We can't see a use case where that is the case, but it's not breaking

Reply:
I understand that this was added because of the guidelines
"Be used to have users select one tile at once."

But that is a design guidelines, we shouldn't do code to prevent it.